src/interface/state_descriptor.hpp

```diff
@@ -80,10 +81,21 @@ struct VarIDHasher {
 /// TODO(JMM): Should this cache be a static member field of
 /// RefinementFunctions_t? That would mean we could avoid
 /// StateDescriptor entirely.
+/// TODO(JMM): The IDs here are not the same as the variable unique
+/// IDs but they maybe could be? We should consider unifying that.
 struct RefinementFunctionMaps {
-  void Register(const Metadata &m) {
+  void Register(const Metadata &m, std::string varname) {
     if (m.IsRefined()) {
       const auto &funcs = m.GetRefinementFunctions();
+      // Guard against uninitialized refinement functions by checking
+      // if the label is the empty string.
+      if (funcs.label().size() == 0) {
+        std::stringstream ss;
+        ss << "Variable " << varname << " registed for refinement, "
+           << "but no prolongation/restriction options found!"
+           << "Please register them with Metadata::RegisterRefinementOps." << std::endl;
+        PARTHENON_THROW(ss);
+      }
```

Comment:
If somehow the registered variable doesn't have a prolongation/restriction op, this will catch it and provide a useful error message. This code should be impossible to reach, but it's in setup so I think it costs us very little to have this check. Especially since we got caught off guard by this once already.